File: 01 Pair and Vector/Vector Punch/d62_q1a_punch.cpp
#include <iostream>
#include <vector>
using namespace std;
vector<string> punch(vector<string> &v, vector<string>::iterator it, int k)
{
    // Copy vector 'v'
    vector<string> NEW(v);
    // Find start and stop erase index
    int start = max(0, int(it-v.begin()-k));
    int stop = min(int(v.size()), int(it-v.begin()+k+1));
    // Erase vector 'NEW'
    NEW.erase(NEW.begin() + start, NEW.begin() + stop);
    // Return new vector
    return NEW;
}
int main() {
    int n,j,k;
    cin >> n >> j >> k;
    vector<string> v(n);
    
    for(int i = 0; i < n; i++) {
        cin >> v[i];
    }
    
    cout << "Result after punch" << endl;
    vector<string> result = punch(v, v.begin() + j, k);
    for(auto &x : result) {
        cout << x << endl;
    }
}

File: 01 Pair and Vector/Vector Punch/d62_q1a_punch.java
import java.util.Scanner;

public class d62_q1a_punch
{
    public static void main(String[] args)
    {
        // Input Value
        Scanner Scanner = new Scanner(System.in);
        int n = Scanner.nextInt();
        int it = Scanner.nextInt();
        int k = Scanner.nextInt();
        String[] items = new String[105];
        for(int i = 0; i < n; i++)
        {
            items[i] = Scanner.next();
        }
        Scanner.close();

        // Find erase range
        int start = Math.max(0, it-k);
        int stop = Math.min(n, it+k);

        // Output
        System.out.println("Result after punch");
        for(int i=0; i<n; i++)
        {
            if(i < start || i > stop)
            {
                System.out.println(items[i]);
            }
        }
    }
}
